src/ResourceManager/ContainerRegistry/Commands.ContainerRegistry/Commands/GetAzureContainerRegistryReplication.cs

```diff
@@ -0,0 +1,112 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.ContainerRegistry
+{
+    [Cmdlet(VerbsCommon.Get, ContainerRegistryReplicationNoun, SupportsShouldProcess = true)]
+    [OutputType(typeof(PSContainerRegistryReplication), typeof(PSContainerRegistryReplication[]))]
+    public class GetAzureContainerRegistryReplication : ContainerRegistryCmdletBase
+    {
+        [Parameter(Position = 0, Mandatory = true, ParameterSetName = ShowReplicationByNameResourceGroupParameterSet, HelpMessage = "Container Registry Replication Name.")]
+        [Parameter(Position = 0, Mandatory = true, ParameterSetName = ShowReplicationByRegistryObjectParameterSet, HelpMessage = "Container Registry Replication Name.")]
+        [ValidateNotNullOrEmpty]
+        [Alias(ReplicationNameAlias)]
+        public string Name { get; set; }
+
+        [Parameter(Position = 1, Mandatory = true, ParameterSetName = ShowReplicationByNameResourceGroupParameterSet, HelpMessage = "Resource Group Name.")]
+        [Parameter(Position = 1, Mandatory = true, ParameterSetName = ListReplicationByNameResourceGroupParameterSet, HelpMessage = "Resource Group Name.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Position = 2, Mandatory = true, ParameterSetName = ShowReplicationByNameResourceGroupParameterSet, HelpMessage = "Container Registry Name.")]
+        [Parameter(Position = 2, Mandatory = true, ParameterSetName = ListReplicationByNameResourceGroupParameterSet, HelpMessage = "Container Registry Name.")]
+        [Alias(ContainerRegistryNameAlias, ResourceNameAlias)]
```

Comment:
This is a general question, in the case where the command is on a child resource of a parent resource, should `ResourceNameAlias` bind with parent resource name (`RegistryName`) or child resource name (`Name`)? I feel like it should be on the child resource name since the child resource is the target here. /cc @cormacpayne